pkg/cachemanager/aggregator/aggregator_test.go

```diff
@@ -215,46 +215,54 @@ func Test_GVKAggregator_Upsert(t *testing.T) {
 }
 
 func Test_GVKAgreggator_Remove(t *testing.T) {
-	t.Run("Remove on empty aggregator", func(t *testing.T) {
-		b := NewGVKAggregator()
-		key := Key{Source: "testSource", ID: "testID"}
-		b.Remove(key)
-	})
-
-	t.Run("Remove non-existing key", func(t *testing.T) {
-		b := NewGVKAggregator()
-		key1 := Key{Source: syncset, ID: "testID1"}
-		key2 := Key{Source: configsync, ID: "testID2"}
-		gvks := []schema.GroupVersionKind{g1v1k1, g1v1k2}
-		b.Upsert(key1, gvks)
-		b.Remove(key2)
-	})
-
-	t.Run("Remove existing key and verify reverseStore", func(t *testing.T) {
-		b := NewGVKAggregator()
-		key1 := Key{Source: syncset, ID: "testID"}
-		gvks := []schema.GroupVersionKind{g1v1k1, g1v1k2}
-		b.Upsert(key1, gvks)
-		b.Remove(key1)
-
-		for _, gvk := range gvks {
-			require.False(t, b.IsPresent(gvk))
-		}
-	})
-
-	t.Run("Remove 1 of existing keys referencing GVKs", func(t *testing.T) {
-		b := NewGVKAggregator()
-		key1 := Key{Source: syncset, ID: "testID"}
-		key2 := Key{Source: configsync, ID: "testID"}
-		gvks := []schema.GroupVersionKind{g1v1k1, g1v1k2}
-		b.Upsert(key1, gvks)
-		b.Upsert(key2, gvks)
-		b.Remove(key1)
-
-		for _, gvk := range gvks {
-			require.True(t, b.IsPresent(gvk))
-		}
-	})
+	tests := []struct {
+		name         string
+		toUpsert     []keyedGVKs
+		keyToRemove  Key
+		expectedGVKs []schema.GroupVersionKind
+	}{
+		{
+			name:        "empty aggregator",
+			keyToRemove: Key{Source: "testSource", ID: "testID"},
+		},
+		{
+			name: "remove non existing key",
+			toUpsert: []keyedGVKs{
+				{key: Key{Source: syncset, ID: "testID1"}, gvks: []schema.GroupVersionKind{g1v1k1, g1v1k2}},
+			},
+			keyToRemove:  Key{Source: configsync, ID: "testID"},
+			expectedGVKs: []schema.GroupVersionKind{g1v1k1, g1v1k2},
+		},
+		{
+			name: "remove existing key",
+			toUpsert: []keyedGVKs{
+				{key: Key{Source: syncset, ID: "testID1"}, gvks: []schema.GroupVersionKind{g1v1k1, g1v1k2}},
+			},
+			keyToRemove: Key{Source: syncset, ID: "testID1"},
+		},
+		{
+			name: "remove overlapping key",
```

Comment:
The name of this test is not accurate, there is no overlap in the gvks.

Since the overlapping thing tests reference counting logic, we should have that. changing `g1v1k2` to `g1v1k1` should suffice.

Reply:
> here is no overlap in the gvks.

hm, both keys reference `g1v1k2`; 

do you want the keys to be fully overlapped not just partially ?

Reply:
nvm didn't see it was the second term overlapping w/the first term